src/build/env.py

```diff
@@ -117,38 +123,44 @@ def _symlink_relative(self, path):  # type: (Optional[str]) -> None
 
     def __enter__(self):  # type: () -> IsolatedEnvironment
         self._path = tempfile.mkdtemp(prefix='build-env-')
-        self._env_vars = {
-            'base': self.path,
-            'platbase': self.path,
-        }
 
-        sys_path = sys.path[1:]
+        if sys.version_info[0] == 2:
+            self._env_vars = {
+                'base': self.path,
+                'platbase': self.path,
+            }
 
-        for path in self.MANIPULATE_PATHS:
-            env_path = self._get_env_path(path)
-            if env_path:
-                sys_path.append(env_path)
+            sys_path = sys.path[1:]
 
-        for path in self._remove_paths:
-            if path in sys_path:
-                sys_path.remove(path)
+            for path in self.MANIPULATE_PATHS:
+                env_path = self._get_env_path(path)
+                if env_path:
+                    sys_path.append(env_path)
 
-        env_scripts = self._get_env_path('scripts')
-        if env_scripts is None:  # pragma: no cover
-            raise RuntimeError('Missing scripts directory in sysconfig')
+            for path in self._remove_paths:
+                if path in sys_path:
+                    sys_path.remove(path)
 
-        exe_path = [env_scripts]
-        if 'PATH' in os.environ:
-            exe_path.append(os.environ['PATH'])
+            env_scripts = self._get_env_path('scripts')
+            if env_scripts is None:  # pragma: no cover
+                raise RuntimeError('Missing scripts directory in sysconfig')
 
-        self._replace_env('PATH', os.pathsep.join(exe_path))
-        self._replace_env('PYTHONPATH', os.pathsep.join(sys_path))
-        self._replace_env('PYTHONHOME', self.path)
-        self._pop_env('PIP_REQUIRE_VIRTUALENV')
+            exe_path = [env_scripts]
+            if 'PATH' in os.environ:
+                exe_path.append(os.environ['PATH'])
 
-        self._symlink_relative(sysconfig.get_path('include'))
-        self._symlink_relative(sysconfig.get_path('platinclude'))
-        self._symlink_relative(sysconfig.get_config_var('LIBPL'))
+            self._replace_env('PATH', os.pathsep.join(exe_path))
+            self._replace_env('PYTHONPATH', os.pathsep.join(sys_path))
+            self._replace_env('PYTHONHOME', self.path)
+
+            self._symlink_relative(sysconfig.get_path('include'))
+            self._symlink_relative(sysconfig.get_path('platinclude'))
+            self._symlink_relative(sysconfig.get_config_var('LIBPL'))
+        else:
+            subprocess.check_call([self.executable, '-m', 'venv', self.path])
```

Comment:
There's no reason to make this a subprocess call. Those are expensive on Windows especially. Use the venv api instead, 

https://docs.python.org/3/library/venv.html#venv.EnvBuilder